### DISCUSSION TITLE
Sync task fails with message "level=fatal msg="Operation has completed with phase: Running""

### DISCUSSION THREAD

**Discussion by iars13** [argoproj/argo-cd#15826] [Q&A] [answered]
Hello everyone, this is my first message, I hope to comply with the site rules. I give you context of my problem. I have a pipeline in openshift where the final step is to run a task that is responsible for synchronizing an application in argo. The final task is the following command: argocd app sync "$(params.application-name)" --revision "$(params.revision)" What I observe is that the time that argo uses to verify It is very short for an application to be in the desired state, so when I run the synchronization command it gives an error because it is already synchronizing, this is where the question comes: Is there a way to increase this remediation time? I tried to use some flag in the command that would send the signal to wait for the synchronization in progress but I couldn't find it. Hopefully you can help me. Greetings.

**Comment by bowale01** [comment]
Increasing remediation time will have to do with adjusting your configurations and i think the solution to this is to configure argo syncwaves which will give priority to an  order to when the different custom resources are applied to the cluster. https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/
let me know if this answers your question

**Reply by iars13** [reply]
Thank you very much for your prompt response, I think I missed specifying that the coincidence in the sync tasks is occurring with the same resource. Let me explain, my pipeline in the previous step updates the configuration in the repo and in the last step requests a synchronization, the problem is that argo is realizing the new desired state very quickly and initializes a synchronization which is colliding with the synchronization manual I am sending. My first idea was to eliminate from my pipeline the last step where I send the manual synchronization but before doing that I am looking for other alternatives because I would have to change many pipelines that I already have configured that way

**Reply by davidmontoyago** [reply]
Hi @iars13, that makes sense. definitely remove the sync step from your pipeline. if the pipeline is already writing to the repo, ArgoCD will take it from there (a la GitOps). it will do so by either polling or via events depending on how you have configured your ArgoCD instance. If you really need to kick the sync from the pipeline you can disable [automatic self-healing](https://argo-cd.readthedocs.io/en/stable/user-guide/auto_sync/#automatic-self-healing)

**Reply by iars13** [reply, marked as answer]
Hello, I was able to do it modifying the script in the task. I did the next changes: 

        sync_status=$(argocd app get "$(params.application-name)" --hard-refresh | grep "Sync Status" | awk '{print $3}')

        if [ "$sync_status" = "Synced" ]; then
          echo "The application is already synchronized."
          argocd app wait "$(params.application-name)" --health "$(params.flags)"
        else
          echo "Starting synchronization..."
          argocd app sync "$(params.application-name)" --revision "$(params.revision)" "$(params.flags)"
          argocd app wait "$(params.application-name)" --health "$(params.flags)"
        fi

Work fine.
Thanks you very much
